arrays/arrays.c

```diff
@@ -21,30 +21,37 @@ typedef struct Array {
  *****/
 Array *create_array (int capacity) {
   // Allocate memory for the Array struct
-
+  Array *array = malloc(sizeof(Array));
   // Set initial values for capacity and count
-
+  array->capacity = cap;
+  array->count = 0;
   // Allocate memory for elements
-
+  array->elements = calloc(cap, sizeof(char *));
+  return array;
 }
 
 
 /*****
  * Free memory for an array and all of its stored elements
  *****/
-void destroy_array(Array *arr) {
-
+void destroy_array(Array *array) {
   // Free all elements
-
+  if (array->elements != NULL)
```

Comment:
You have to go through and free each individual element e.g. `arr->elements[i]`, assuming you alloc'd memory for it when you made a copy of it in `arr_insert()` and `arr_append()`